### PR TITLE
Unblock BinaryFormatter tests on UAPAOT

### PR DESCRIPTION
**Pull request by morganbr**
Fixes to unblock BinaryFormatter tests on UAPAOT. Previously, these issues caused some of the generator methods for Theories to fail so most tests weren't running. With this change, we run an additional ~2000 tests. A few hundred of those will fail, but we should get more honest coverage.

### CODE REVIEW

**Comment by morganbr** [comment]
CC @dotnet/corert-contrib

**Comment by ViktorHofer** [comment]
thanks @morganbr!

File: src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/DeserializationEventHandler.cs [hunk 1]
```diff
@@ -6,5 +6,5 @@ namespace System.Runtime.Serialization
 {
     internal delegate void DeserializationEventHandler(object sender);
 
-    internal delegate void SerializationEventHandler(StreamingContext context);
+    public delegate void SerializationEventHandler(StreamingContext context);
```
Comment (ViktorHofer) [inline src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/DeserializationEventHandler.cs]: Isn't this an API addition and should be reviewed?

**Comment by AtsushiKan** [comment]
Maybe you're planning this for a later commit, but most of the serialization tests have now been rewritten to use this common helper and until these two lines are gone, the tests aren't really enabled...

https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Runtime/Serialization/Formatters/BinaryFormatterHelpers.cs#L18

https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Runtime/Serialization/Formatters/BinaryFormatterHelpers.cs#L33

**Comment by morganbr** [comment]
@AtsushiKan, thanks for highlighting that. I don't believe the S.R.Serialization.Formatters.Tests suite uses those helpers, but you're right that removing those lines will be very important once the other tests are in good shape.

**Pull request merged**